src/interface/mod.rs

```diff
@@ -150,3 +150,53 @@ pub trait Resolver<K = Spur> {
         self.len() == 0
     }
 }
+
+impl<T, K> Resolver<K> for &T
+where
+    T: Resolver<K>,
+{
+    fn resolve<'a>(&'a self, key: &K) -> &'a str {
```

Comment:
I just realised maybe these should all have `#[cfg_attr(feature = "inline-more", inline)]` on them?